lib/portage/_emirrordist/DeletionIterator.py

```diff
@@ -20,17 +20,25 @@ def __iter__(self):
 		deletion_db = self._config.deletion_db
 		deletion_delay = self._config.options.deletion_delay
 		start_time = self._config.start_time
-		distfiles_set = set(os.listdir(self._config.options.distfiles))
+		distfiles_set = set()
+		for layout in self._config.layouts:
+			distfiles_set.update(layout.get_filenames(distdir))
 		for filename in distfiles_set:
-			try:
-				st = os.stat(os.path.join(distdir, filename))
-			except OSError as e:
-				logging.error("stat failed on '%s' in distfiles: %s\n" %
-					(filename, e))
-				continue
-			if not stat.S_ISREG(st.st_mode):
+			# require at least one successful stat()
+			for layout in self._config.layouts:
+				try:
+					st = os.stat(
+							os.path.join(distdir, layout.get_path(filename)))
+				except OSError as e:
+					logging.error("stat failed on '%s' in distfiles: %s\n" %
+						(filename, e))
```

Comment:
This will produce some spurious error messages if all layouts do not contain exactly the same files. Do we expect all layouts to be consistent with each other at all times?

Reply:
I was wondering about it but couldn't come up with a good resolution. But yes, we expect them to be consistent except for a short initial propagation period.

What do you suggest? Storing the exception for the first layout, and logging it only if none of the files stat()ed correctly?

Reply:
Yeah, storing first exception sounds good.